designer/server/src/main/scala/pl/touk/nussknacker/ui/process/processingtypedata/ProcessingTypeDataProvider.scala

```diff
@@ -24,8 +24,6 @@ import pl.touk.nussknacker.ui.security.api.LoggedUser
  */
 trait ProcessingTypeDataProvider[+T, +C] {
```

Comment:
it looks like the name is too generic. I can find in the interface references to ProcessingType but I don't know what is the `Data` in this context. Moreover, we can find references to LoggedUser in the methods. Why? It's not obvious how T refers to ProcessingType and LoggedUser.

Reply:
`ProcessingType` is a kind of `ApplicationContext` - similar to this one from `Spring`. It could be said that we have "applications inside application". One `ProcessingType` can't see data from other `ProcessingType's`. Services run inside one `ProcessingType` scope behave differently from services run in another scope.

`Data` is the generic name here by purpose. The purpose of this interface was to make accessible any data, hidden by this interface. You are starting from the whole application context which is `ProcessingTypeData` class (probably name should contain "Root"). Then you want to pass somewhere only a piece of information inside it, so you do some transformation, the same for each context.

- `T` is these `Data` available for certain `ProcessingType`
- `C` is `CombinedData`. We have a limited set of data which is a kind of intersection between all `ProcessingTypes`. We can discuss this further because it is a longer story.

We have `LoggedUser` because our `ProcessingTypes` are associated with `Category` e.g. Fraud Detection, Marketing. Given user has access to certain categories - see `LoggedUser.can`. `ProcessingTypeDataProvider` is a kind of gateway to all around `ProcessingType` so it is the first place where we can authorize user. Almost every endpoint is inside `ProcessingType`, and in each of them we at the beginning authorize user access to given `ProcessingType`

Reply:
1. Done
2. Done
3. I'd rather don't want to break this convention. We can discuss the convention around implicit parameters in the whole project separately.